### PR TITLE
Add badge for year-long #12in23 challenge

### PR DESCRIPTION
**Pull request by ErikSchierboom**
See https://forum.exercism.org/t/wrinkle-for-12in23-include-exercise-if-previously-completed-in-all-tracks/5012

### CODE REVIEW

File: app/commands/user/challenges/featured_exercises_progress_12_in_23.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+      solved_before_23 = solved_in_featured_tracks.select { |_, year| year < 2023 }
+      next [solved_before_23.keys.first, exercise_slug] if solved_before_23.size == track_slugs.size
+    end
+  end
+
+  private
+  memoize
+  def solutions
+    user.solutions.published.
```
Comment (ErikSchierboom) [inline app/commands/user/challenges/featured_exercises_progress_12_in_23.rb:23]: I was slightly confused about whether the solution has to be published or not for it to count.

The "The year-long #12in23 badge" section has this description:

> At the end of the year we'll be awarding an exclusive badge to anyone who completes all the featured exercises in the correct paradigms. You can complete the exercise any time during 2023. Each valid exercise will be shown below

No mention of published there, but the exercises it lists are checked via published (which I copied).

File: app/commands/user/challenges/featured_exercises_progress_12_in_23.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      group_by(&:first).
+      transform_values { |solutions| solutions.map { |solution| [solution[1], solution[2].year] }.to_h }
+  end
+
+  memoize
+  def featured_exercises
+    (FEBRUARY_EXERCISES.map { |e| [e, FEBRUARY_TRACKS] } +
+     MARCH_EXERCISES.map { |e| [e, MARCH_TRACKS] } +
+     APRIL_EXERCISES.map { |e| [e, APRIL_TRACKS] } +
+     MAY_EXERCISES.map { |e| [e, MAY_TRACKS] }).to_h
```
Comment (iHiD) [inline app/commands/user/challenges/featured_exercises_progress_12_in_23.rb]:
```suggestion
    (
    FEBRUARY_EXERCISES.map { |e| [e, FEBRUARY_TRACKS] } +
     MARCH_EXERCISES.map { |e| [e, MARCH_TRACKS] } +
     APRIL_EXERCISES.map { |e| [e, APRIL_TRACKS] } +
     MAY_EXERCISES.map { |e| [e, MAY_TRACKS] }
     ).to_h
```

(With correct indenting)

**Review by iHiD — APPROVED** [review]
Nice!

I guess this can get merged and just added to each month?

**Comment by ErikSchierboom** [comment]
> I guess this can get merged and just added to each month?

Yes, because we don't award the badge yet.

**Comment by iHiD** [comment]
Cool. Do it.

**Pull request merged**